src/darker/tests/test_utils.py

```diff
@@ -229,29 +229,35 @@ def test_textdocument_from_str(
     dict(content=b'\xef\xbb\xbfprint("touch\xc3\xa9")\n', expect="utf-8-sig"),
     dict(content=b'# coding: iso-8859-1\n"touch\xe9"\n', expect="iso-8859-1"),
 )
-def test_textdocument_from_file_detect_encoding(tmp_path, content, expect):
-    """TextDocument.from_file() detects the file encoding correctly"""
+def test_textdocument_from_file_or_bytes_detect_encoding(tmp_path, content, expect):
+    """TextDocument.from_file()/.from_bytes() detects the file encoding correctly"""
     path = tmp_path / "test.py"
     path.write_bytes(content)
 
     textdocument = TextDocument.from_file(path)
 
     assert textdocument.encoding == expect
 
+    textdocument = TextDocument.from_bytes(content)
+    assert textdocument.encoding == expect
```

Comment:
In these tests, if `from_file()` fails, `from_bytes` won't be tested at all. Again, that's not optimal for a unit test.

This could be improved e.g. by doing another parameterization by the `.from_*()` method name, or by using [pytest-assume](https://github.com/astraw38/pytest-assume).